### DISCUSSION TITLE
Do you have a recommended flow, for fetching a signature when doing signed uploads?

### DISCUSSION THREAD

**Discussion by alexbjorlig** [uploadcare/file-uploader#406] [blocks]
The most ideal thing (if you ask me) would be:

1. User click's un upload
2. There is a hook where we can fetch a signed token from our API (and await it), and pass it to blocks
3. Profit 😀

Do you have an example of the above?

**Comment by loookashow** [comment]
Yes, your option is great, but we cannot await data provided by third-party API inside of the file uploading process, because we don't know if that API works fast and correctly. So, for example, you can get a newly signed token on page loading or if you use inline mode inserted in your own modal window - when the modal window is opened.

**Comment by loookashow** [comment]
BTW, you are also able to write some additional metadata for uploading files, when a page or a modal is initialized, e.g.:

```js
if (this.dialogVisible) {
          await connectBlocksFrom ('https://unpkg.com/@uploadcare/blocks@0.19.0/web/blocks-browser.min.js')
            .then (() => {
              let uploaderCtx = this.$refs.uploaderctx
              let uploader = this.$refs.uploader
              this.$api.files.getSignature ()
                .then (_ => {
                  uploaderCtx.setUploadMetadata ({
                    company: _.data.company,
                    user: _.data.user,
                    project: _.data.project
                  })
                  uploader.style.setProperty ('--cfg-secure-signature', `"${_.data.signature}"`)
                  uploader.style.setProperty ('--cfg-secure-expire', `"${_.data.expire}"`)
                })
            })
        }
```
